### ISSUE TITLE
Problems on windows

### ISSUE THREAD

**Issue by vors** [Gazler/githug#186]
level 37 (push_branch)
It uses [unix-specific command wc](https://github.com/Gazler/githug/blob/master/levels/push_branch.rb)

The error message:
```
'wc' is not recognized as an internal or external command,
operable program or batch file.
Sorry, this solution is not quite right!
```

**Comment by vors** [comment]
Same with level 39 (fetch)

**Comment by vors** [comment]
Name: reorder
Level: 47
Difficulty: ****

You have committed several times but in the wrong order. Please reorder your commits.


C:/Ruby21-x64/lib/ruby/2.1.0/fileutils.rb:1163:in `utime': Permission denied @ utime_failed - file3 (Errno::EACCES)
        from C:/Ruby21-x64/lib/ruby/2.1.0/fileutils.rb:1163:in `block in touch'
        from C:/Ruby21-x64/lib/ruby/2.1.0/fileutils.rb:1160:in `each'
        from C:/Ruby21-x64/lib/ruby/2.1.0/fileutils.rb:1160:in `touch'
        from (eval):15:in `block in setup'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/lib/githug/level.rb:90:in `call'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/lib/githug/level.rb:90:in `setup_level'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/lib/githug/game.rb:52:in `level_bump'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/lib/githug/game.rb:21:in `play_level'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/lib/githug/cli.rb:14:in `play'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/thor-0.14.6/lib/thor/invocation.rb:118:in `invoke_task'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/thor-0.14.6/lib/thor.rb:263:in `dispatch'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/thor-0.14.6/lib/thor/base.rb:389:in `start'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/githug-0.4.3/bin/githug:4:in `<top (required)>'
        from C:/Ruby21-x64/bin/githug:23:in `load'
        from C:/Ruby21-x64/bin/githug:23:in `<main>'

**Comment by Gazler** [comment]
@vors Thanks for raising all these. I did used to support Windows when I had easy access to a Windows machine, I'll try to replace these tools with pure ruby alternatives.

I'm not sure what the issue on the reorder level is, maybe a lock on the file?

The files/lines in question are:

* push_branch - https://github.com/Gazler/githug/blob/ab0a9692e900bbe742e9bf248e4a0836dd376261/levels/push_branch.rb#L63
* fetch - https://github.com/Gazler/githug/blob/ab0a9692e900bbe742e9bf248e4a0836dd376261/levels/fetch.rb#L55
* reorder - https://github.com/Gazler/githug/blob/ab0a9692e900bbe742e9bf248e4a0836dd376261/levels/reorder.rb

**Comment by vors** [comment]
To prevent future regressions you may want to have appveyor build script in repo. Appveyor is a travis-ci for windows. It's free for open source and use yml.

**Comment by Gazler** [comment]
@vors I have created a new branch at https://github.com/Gazler/githug/tree/fix/windows for windows fixes. I would appreciate it if you could either run the tests or manually verify the levels in question using the following steps:

To run the tests:

```
git fetch git@github.com:Gazler/githug.git
git checkout fix/windows
rake
```

To test manually:

```
git fetch git@github.com:Gazler/githug.git
git checkout fix/windows
rake
gem install pkg/githug-0.4.3.gem
cd YOUR_GITHUG_DIRECTORY
githug reset fetch
git fetch
githug
githug reset push_branch
git push origin test_branch
githug
```

I am still not sure about the reorder one, I would appreciate it if you could just reset the level and try again for that one.

With regards to Appveyor, am trying to set it up now.

@abudarian - you may be interested in this as well.

Thanks

**Comment by vors** [comment]
Just tried test with rake, got problems for all levels:
```
...Exiting
..********************************************************************************
*                                    Githug                                    *
********************************************************************************
...********************************************************************************
*                                    Githug                                    *
********************************************************************************
...............................this is hint 1
...................................FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
An error occurred in an after(:all) hook.
  TypeError: no implicit conversion of nil into String
  occurred at D:/dev/githug/spec/githug_spec.rb:29:in `chdir'



Failures:

  1) The Game solves the init level
     Failure/Error: Dir.chdir("/tmp")
     Errno::ENOENT:
       No such file or directory @ dir_chdir - /tmp
     # ./spec/githug_spec.rb:23:in `chdir'
     # ./spec/githug_spec.rb:23:in `block (2 levels) in <top (required)>'

  2) The Game solves the config level
     Failure/Error: Dir.chdir("/tmp")
     Errno::ENOENT:
       No such file or directory @ dir_chdir - /tmp
     # ./spec/githug_spec.rb:23:in `chdir'
     # ./spec/githug_spec.rb:23:in `block (2 levels) in <top (required)>'
...
```

**Comment by Gazler** [comment]
@vors The tests actually use many linux commands to fetch the commit hashes for certain levels. I have tried to get Appveyor working, but didn't have much luck with it.

I have however fixed all 3 issues that you have raised and released githug v0.4.4

The issue for the reorder level was that the previous level didn't close one of the files in the solution.

Thanks for raising the issues!

cc @abudarian

**Comment by vors** [comment]
That's great! Thank you.
I can try to provide windows alternative for such commands with PowerShell.
